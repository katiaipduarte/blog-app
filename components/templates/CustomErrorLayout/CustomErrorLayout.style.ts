import styled from 'styled-components';

export const CustomErrorContainer = styled.div`
  display: flex;
  justify-content: flex-start;
  flex-direction: column;

  .http-code {
    padding: 2rem 0;

    h1 {
      font-size: 3.5rem;
      font-weight: 800;
      line-height: 2.25rem;
    }
  }

  .message {
    h2 {
      margin-bottom: 1rem;
    }

    p {
      font-size: 1rem;
      margin-bottom: 2rem;
    }

    button {
      display: block;
      margin: 0 auto;
    }
  }

  @media all and (min-width: 700px) {
    justify-content: center;
    align-items: center;
    flex-direction: row;
    margin-top: 6rem;

    .http-code {
      h1 {
        font-size: 6rem;
        line-height: 3rem;
        border-right: 2px solid ${({ theme }) => theme.text};
        padding: 0 1.5rem;
      }
    }

    .message {
      max-width: 28rem;
      padding: 0 1.5rem;

      h2 {
        margin-bottom: 2rem;
      }

      button {
        margin: 0;
      }
    }
  }
`;
