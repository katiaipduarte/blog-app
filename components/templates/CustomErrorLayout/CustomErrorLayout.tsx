import React from 'react';
import Link from 'next/link';

import { CustomErrorContainer } from './CustomErrorLayout.style';

type Props = {
  errorCode: number;
  message: string;
};

const CustomErrorLayout = (props: Props): JSX.Element => {
  const { errorCode, message } = props;

  return (
    <CustomErrorContainer>
      <div className="http-code">
        <h1 tabIndex={0} aria-label={`Error ${errorCode}`}>
          {errorCode}
        </h1>
      </div>
      <div className="message">
        <h2 tabIndex={0}>{message}</h2>
        <p role="text" tabIndex={0}>
          Don't worry just click on the following button and you can go back to
          our homepage!
        </p>
        <Link href="/">
          <button
            className="full"
            aria-label="Go back to homepage"
            type="button"
          >
            Go back to homepage.{' '}
          </button>
        </Link>
      </div>
    </CustomErrorContainer>
  );
};

export default CustomErrorLayout;
