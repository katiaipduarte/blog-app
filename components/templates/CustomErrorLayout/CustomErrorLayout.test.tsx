import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import CustomErrorLayout from './CustomErrorLayout';

describe('the <CustomErrorLayout /> component', () => {
  it('should render custom error layout with code and message', () => {
    const tree = renderer
      .create(<CustomErrorLayout errorCode={404} message="Unit test" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
