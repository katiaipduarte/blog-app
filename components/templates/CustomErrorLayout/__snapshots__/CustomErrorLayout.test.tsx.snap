// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <CustomErrorLayout /> component should render custom error layout with code and message 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .http-code {
  padding: 2rem 0;
}

.c0 .http-code h1 {
  font-size: 3.5rem;
  font-weight: 800;
  line-height: 2.25rem;
}

.c0 .message h2 {
  margin-bottom: 1rem;
}

.c0 .message p {
  font-size: 1rem;
  margin-bottom: 2rem;
}

.c0 .message button {
  display: block;
  margin: 0 auto;
}

@media all and (min-width:700px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-top: 6rem;
  }

  .c0 .http-code h1 {
    font-size: 6rem;
    line-height: 3rem;
    border-right: 2px solid;
    padding: 0 1.5rem;
  }

  .c0 .message {
    max-width: 28rem;
    padding: 0 1.5rem;
  }

  .c0 .message h2 {
    margin-bottom: 2rem;
  }

  .c0 .message button {
    margin: 0;
  }
}

<div
  className="c0"
>
  <div
    className="http-code"
  >
    <h1
      aria-label="Error 404"
      tabIndex={0}
    >
      404
    </h1>
  </div>
  <div
    className="message"
  >
    <h2
      tabIndex={0}
    >
      Unit test
    </h2>
    <p
      role="text"
      tabIndex={0}
    >
      Don't worry just click on the following button and you can go back to our homepage!
    </p>
    <button
      aria-label="Go back to homepage"
      className="full"
      onClick={[Function]}
      onMouseEnter={[Function]}
      type="button"
    >
      Go back to homepage.
    </button>
  </div>
</div>
`;
